validator/validator.go

```diff
@@ -8,7 +8,7 @@ import (
 
 type AddErrFunc func(options ...ErrorOption)
 
-type ruleFunc func(observers *Events, addError AddErrFunc)
+type ruleFunc func(observers *Events, validateOption ValidateOption, addError AddErrFunc)
```

Comment:
```suggestion
type ruleFunc func(observers *Events, addError AddErrFunc, options ...ValidateOption)
```
Currently people are using the AddRule which takes a function as an argument that uses this `ruleFunc` signature, so your change here would break backwards compatibility unless you made it variadic.